src/main/java/am/itspace/sweetbakerystore/controller/web/ShopController.java

```diff
@@ -1,16 +1,61 @@
 package am.itspace.sweetbakerystore.controller.web;
 
+import am.itspace.sweetbakerystore.entity.Product;
+import am.itspace.sweetbakerystore.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.MediaType;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 @Controller
+@RequiredArgsConstructor
 public class ShopController {
+    private final ProductService productService;
+
     @GetMapping(value = "/shop")
-    public String shop() {
+    public String shop(ModelMap modelMap,
+                       @RequestParam("page") Optional<Integer> page,
+                       @RequestParam("size") Optional<Integer> size) {
```

Comment:
Use @PgaeDefault Pageable pageable instead of page and size

Reply:
Done